client_option.go

```diff
@@ -0,0 +1,63 @@
+package gax
+
+type ClientOption interface {
+	Resolve(*clientSettings)
```

Comment:
I'd unexport this method. There's no need for it to be visible in package docs, and other packages can't implement this interface anyway.